primo/utils/opt_utils.py

```diff
@@ -211,3 +216,46 @@ def budget_slack_variable_scaling(
         ) / model_inputs.budget
         budget_sufficient = True
     return scaling_budget_slack, budget_sufficient
+
+
+class OptimizationException(Exception):
+    """Exception for the optimization"""
+
+
+def optimization_results_handler(results: SolverResults, model: pyo.ConcreteModel):
+    """
+    A function to handle the various results returned by an optimization solve
+
+    Parameters
+    ----------
+    results : SolverResults
+        Result of an optimization solve
+
+    model : pyo.ConcreteModel
+        Pyomo model being solved
+
+    Returns
+    -------
+        1 : Optimal termination
```

Comment:
Can we wrap the statuses in an enum class or atleast define them as global parameters somewhere? That way, we are not left with magic numbers (1 or 2) in other places in the code